src/lib.rs

```diff
@@ -159,6 +161,47 @@ pub enum SanType {
 	IpAddress(IpAddr),
 }
 
+#[cfg(feature = "x509-parser")]
+fn ip_addr_from_octets(octets: &[u8]) -> Result<IpAddr, RcgenError> {
+	let len = octets.len();
+
+	let ip = match len {
+		4 => Ipv4Addr::new(
+			octets[0],
+			octets[1],
+			octets[2],
+			octets[3]
+		)
+			.into(),
+		16 => {
+			let o: [u8; 16] = [
```

Comment:
Can you use `= octets.try_into().unwrap()`? This should do it I think. The unwrap is safe because the size is known.

Reply:
Doesn't compile:

`the trait From<&[u8]> is not implemented for Ipv4Addr`

Reply:
There's a `nightly` api that may help with this in the future:

https://doc.rust-lang.org/std/primitive.slice.html#method.split_array_ref

Reply:
Yeah you'd have to first convert the slice into an array reference, and then convert that array reference into an array, and *then* you would have to convert that into an `Ipv4Addr`. No nightly features needed, everything stable since over a year ago.

Reply:
lol, that stretched my rust legs, but I figured out how to do it and updated the PR.